app/screens/user_profile/user_profile.js

```diff
@@ -60,17 +97,27 @@ export default class UserProfile extends PureComponent {
         return null;
     };
 
-    sendMessage = () => {
-        const {actions, myPreferences, navigator, user} = this.props;
+    sendMessage = async () => {
+        const {actions, currentDisplayName, myPreferences, navigator, user} = this.props;
+
+        // save the current channel display name in case it fails
+        const currentChannelDisplayName = currentDisplayName;
+
         actions.setChannelDisplayName(displayUsername(user, myPreferences));
-        actions.handleSendMessage(user.id);
-        navigator.pop({
-            animated: true
-        });
+
+        const result = await actions.makeDirectChannel(user.id);
+        if (result) {
+            navigator.pop({
+                animated: true
+            });
+        } else {
+            actions.setChannelDisplayName(currentChannelDisplayName);
+            this.setState({error: true});
```

Comment:
Shouldn't we just fire the alert here? Setting the state and then checking it in componentWillReceiveProps is doing this in a really roundabout way